### PR TITLE
Sidemenu: Various accessibility improvements

### PR DESCRIPTION
**Pull request by ashharrison90**
**What this PR does / why we need it**:

various accessibility improvements:
- when no url is present, use `<button>` instead of an `<a>` element. this can then be tabbed to. e.g. the switch organization button or the keyboard shortcuts button: 
image
or the search dashboards header: 
image
- some css needed to render the buttons the same as the anchor elements.
- added `alt` attributes to any elements with images. e.g. the profile sidemenuitem
- hide the entire `sidemenu-link` element when in mobile view. this removes some empty elements that could still be accessed via tab like this:
image

note: this is not the full accessibility changes needed. we still need some input from ux/design on what the sidemenuitems will look like before we do a full deep dive on accessibility.

**Which issue(s) this PR fixes**:

For https://github.com/grafana/grafana/issues/38507

**Special notes for your reviewer**:

### CODE REVIEW

**Review by Clarity-89 — APPROVED** [review]
Looks good!

**Review by mckn — APPROVED** [review]
Looks really good!

**Comment by jackw** [comment]
Great work on this Ash. 🚀  

I have a question around the `dropdown-menu` css classes and the additions of button element selectors in this PR. I could be worrying about nothing but the `component/_dropdown.scss` file name and the generic class name has me thinking...

As an element selector is used for the button, if there are plugins that already make use of `.dropdown-menu` and has button elements inside it this change will cause the styles in this PR to cascade down and potentially break the plugins styles. As such I'm wondering if the button selectors would be "safer" with class names?

**Comment by ashharrison90** [comment]
@jackw hmmmmm good question. 🤔 

i _think_ it's pretty safe given the use of the child combinator `>` in the css. it will only affect `button`s that are immediate descendants of `li`, which are pretty much guaranteed to want to be styled like this.

it probably would be safer as a class name, but then we have the opposite problem of someone having to remember to add this arbitrary class name whenever adding a button inside a `.dropdown-menu`. 

my immediate inclination is to leave it for now and see if anyone shouts 😅  wdyt?

semi related question: i know we expose `@grafana/ui` as explicit components for plugin developers to use, but are we also expecting any of the components in `public/app/core/components` to be used by plugin devs as well?

**Comment by mckn** [comment]
Good point @jackw! Regarding what was/is exposed to the "old" plugins API you can view it here:

https://github.com/grafana/grafana/blob/main/public/app/features/plugins/plugin_loader.ts

**Comment by mckn** [comment]
So I guess it depends on if any of the components are being used in any of the core components that are being exposed to plugins.

**Comment by mckn** [comment]
Sorry for not thinking about this earlier during my review btw.

**Comment by mckn** [comment]
Yeah, that is how I read it as well. @torkelo do you want to just verify it?

**Comment by jackw** [comment]
> so if i'm reading that right (and i might not be...) i don't see anywhere public/app/core/components being exposed. if so, does that mean the only consumers of e.g. dropdown-menu is grafana internally?

I don't know this part of the codebase or angular very well but the only thing I can see is [this gfDropdown directive](https://github.com/grafana/grafana/blob/main/public/app/angular/misc.ts#L154-L208) has a template that uses the `dropdown-menu` class that looks to be attached to the coreModule which is exposed in [plugin_loader](https://github.com/grafana/grafana/blob/main/public/app/features/plugins/plugin_loader.ts#L141).

**Comment by torkelo** [comment]
dropdown-menu class is legacy and something we should try to not use (heritage all the way back to bootstrap) 

Yes side menu components is not exposed to plugins, and they should not be

File: public/app/core/components/sidemenu/SideMenuDropDown.tsx [hunk 1]
```diff
@@ -21,15 +21,19 @@ const SideMenuDropDown = ({
...
+    font-size: inherit;
+  `;
   const headerContent = <span className="sidemenu-item-text">{headerText}</span>;
   const header = headerUrl ? (
     <Link href={headerUrl} onClick={onHeaderClick} className="side-menu-header-link">
       {headerContent}
     </Link>
   ) : (
-    <a onClick={onHeaderClick} className="side-menu-header-link">
+    <button onClick={onHeaderClick} className={cx(resetButtonStyles, 'side-menu-header-link')}>
```
Comment (torkelo) [inline public/app/core/components/sidemenu/SideMenuDropDown.tsx:34]: This seems hacky, we should update the side-menu-header-link style (or migrate it to emotion) not just add some emotion styles on top

**Review by torkelo — COMMENTED** [review]
Not sure about the reset button styles , and why this is a dynamic cx addition , cx should be used for dynamic style composition based on component state/props , but this looks different, like it should just be part of the component styles with no dynamic composition needed

**Comment by ashharrison90** [comment]
yeah, you're right. I can do that on monday. I can also migrate the sidemenu styles to emotion as well I think. the dropdown ones are a little trickier cause they're shared across multiple components.

**Pull request merged**